### ISSUE TITLE
Add hue-degree-notation

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#4721]
> What is the problem you're trying to solve?

I'd like to enforce a consistent hue notation.

From https://github.com/stylelint/stylelint/issues/4718#issuecomment-619336037:

"- consider being opinionated about using the `deg` suffix where applicable, e.g. in the first parameter for `hsl()` and the last non-alpha parameter for `lch()`. Again, because this clarifies that the range lies between `0-360`."

References:

- [Spec](https://www.w3.org/TR/css-color-4/#typedef-hue)
- [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#HSL_colors)

> What solution would you like to see?

A new rule that lets me enforce either a `<number>` or `<angle>` notation for `<hue>`.

Primary option: `string: "number"|"angle"`

With `"number"` the following are valid:

```css
a { 
  color: hsl(198 28% 50%);
  color: lch(56.29% 19.86 236.62 / 50%);
  ..
}
```

With `"angle"` the following are valid:

```css
a { 
  color: hsl(198deg 28% 50%);
  color: lch(56.29% 19.86 1turn / 50%);
  hsl(4.71239rad, 60%, 70%);
  ..
}
```

I can, if desired, enforce the [type of angle](https://www.w3.org/TR/css3-values/#angle-value) using [`unit-whitelist`](https://stylelint.io/user-guide/rules/unit-whitelist).

Any thoughts about the use case, name or options?

**Comment by mathiasbynens** [comment]
The above sounds good, but in addition, I'd want to enforce a specific type of angle notation. Maybe add `angle-deg`, `angle-grad`, `angle-rad`, `angle-turn` options? This way autofix behavior would be unambiguous.

**Comment by jeddy3** [comment]
> This way autofix behavior would be unambiguous.

That's a good catch. 

> I'd want to enforce a specific type of angle notation.

I think we should cater to that in a separate rule to be flexible. The issue in hand is clarifying that the range lies between `0-360`.

From [the hue spec](https://www.w3.org/TR/css-color-4/#typedef-hue):

> The first argument specifies the hue. ... Because this value is so often given in degrees, the argument can also be given as a number, which is interpreted as a number of degrees.

Or said another way; when I use degrees for hues I want to enforce:

- using `deg` units to be explicit
- not using `deg` units for shorthand convenience

If we were to add another rule that enforces degrees, radians, turns or gradians for every hue, it would work [in conjunction](https://stylelint.io/user-guide/rules/combine) with this rule and they would cater to both these use cases:

- I want to only use gradians/degrees/turns/radians for hues (autofixable)
- I don't always want to use degrees for hues (not autofixable), but when I do I don't want to use the `deg` unit (autofixable)

The latter plays well with broader (but not autofixable) [`*-blacklist` and `*-whitelist` rules](https://stylelint.io/user-guide/rules/list#limit-language-features), which when used in conjunction with this rule would cater to:

- I want to use either degrees or radians for hues (not autofixable), but when I use degrees I want to use the `deg` unit (autofixable)

With this narrower scope in mind, we can remove the ambiguity by naming the rule `hue-degree-notation`.

Rule blueprint: 

- Name: `hue-degree-notation`
- Primary option: `"number"|"angle"`
- Secondary options: None
- Message: `Expected "${unfixed}" to be "${fixed}"`, e.g. 'Expected "100" to be "100deg" (hue-degree-notation)'
- Description: "Specify number or angle notation for degree hues."
- Extended description: "Because hues are so often given in degrees, a hue can also be given as a number, which is interpreted as a number of degrees".
- Section: ["Limit language features" -> "Color"](https://stylelint.io/user-guide/rules/list#color-1)

**Comment by jeddy3** [comment]
As there are no objections, I'll label this rule as good-to-go.

Please comment below if anyone would like to [contribute](https://stylelint.io/contributing) this rule to stylelint.

There are [steps on how to add a new rule](https://stylelint.io/developer-guide/rules#add-a-rule) in the Developer guide. The [color-named](https://github.com/stylelint/stylelint/tree/master/lib/rules/color-named) rule would be a good rule to base this new rule off as it also deals with colours and uses the value-parser.
